arangod/Pregel/Algos/AIR/AccumulatorOptions.cpp

```diff
@@ -0,0 +1,387 @@
+template <typename V>
+using my_vector = std::vector<V>;
+template <typename V>
+using my_unordered_set = std::unordered_set<V>;
+
+bool isValidAccumulatorOptions(const AccumulatorOptions& options);
+
+/* clang-format off */
+
+constexpr const char accumulatorType_max[] = "max";
+constexpr const char accumulatorType_min[] = "min";
+constexpr const char accumulatorType_sum[] = "sum";
+constexpr const char accumulatorType_and[] = "and";
+constexpr const char accumulatorType_or[] = "or";
+constexpr const char accumulatorType_store[] = "store";
+constexpr const char accumulatorType_list[] = "list";
+constexpr const char accumulatorType_custom[] = "custom";
+
+using accumulator_type_deserializer = enum_deserializer<AccumulatorType,
+    enum_member<AccumulatorType::MIN, values::string_value<accumulatorType_min>>,
+    enum_member<AccumulatorType::MAX, values::string_value<accumulatorType_max>>,
+    enum_member<AccumulatorType::SUM, values::string_value<accumulatorType_sum>>,
+    enum_member<AccumulatorType::AND, values::string_value<accumulatorType_and>>,
+    enum_member<AccumulatorType::OR, values::string_value<accumulatorType_or>>,
+    enum_member<AccumulatorType::STORE, values::string_value<accumulatorType_store>>,
+    enum_member<AccumulatorType::LIST, values::string_value<accumulatorType_list>>,
+    enum_member<AccumulatorType::CUSTOM, values::string_value<accumulatorType_custom>>
+>;
+
+constexpr const char accumulatorValueType_doubles[] = "doubles";
+constexpr const char accumulatorValueType_ints[] = "ints";
+constexpr const char accumulatorValueType_strings[] = "strings";
+constexpr const char accumulatorValueType_double[] = "double";
+constexpr const char accumulatorValueType_int[] = "int";
+constexpr const char accumulatorValueType_string[] = "string";
+constexpr const char accumulatorValueType_bool[] = "bool";
+constexpr const char accumulatorValueType_slice[] = "slice";
+constexpr const char accumulatorValueType_any[] = "any";
+
+using accumulator_value_type_deserializer = enum_deserializer<AccumulatorValueType,
+    enum_member<AccumulatorValueType::DOUBLE, values::string_value<accumulatorValueType_doubles>>,
+    enum_member<AccumulatorValueType::INT, values::string_value<accumulatorValueType_ints>>,
+    enum_member<AccumulatorValueType::STRING, values::string_value<accumulatorValueType_strings>>,
+    enum_member<AccumulatorValueType::DOUBLE, values::string_value<accumulatorValueType_double>, true>,
+    enum_member<AccumulatorValueType::INT, values::string_value<accumulatorValueType_int>, true>,
+    enum_member<AccumulatorValueType::STRING, values::string_value<accumulatorValueType_string>, true>,
+    enum_member<AccumulatorValueType::BOOL, values::string_value<accumulatorValueType_bool>>,
+    enum_member<AccumulatorValueType::SLICE, values::string_value<accumulatorValueType_slice>, true>,
+    enum_member<AccumulatorValueType::SLICE, values::string_value<accumulatorValueType_any>>
+>;
```

Comment:
I think the old variants can be removed now - the documentation only lists _any, int, double, bool, string_. The other ones, i.e. _doubles, ints, strings, slice_ can be removed.

Also note that it seems to be bogus where the `hidden` argument is set; sometimes for the up-to-date variants (double, int, string) and once for the outdated one (slice). But that can always be omitted here if the old values are removed.

Reply:
Removed not needed / unused types. Removed "slice" as a type and changed to "any". Adjusted Pregelator and documentation.